main.cc

```diff
@@ -62,9 +62,23 @@ int main(int argc, char *argv[]) {
     while (!input.eof()) {
         line_no++;
         getline(input, line);
+
+	// Strip any comments that start with ; to the end of the line
+	size_t comment_start = line.find(";");
+	if (comment_start != std::string::npos) {
+		line.replace(comment_start, line.length(), "");
```

Comment:
It is probably cheaper to just say `line.resize(comment_start)`;. That way it just adapts the length without any potential re-allocation (which a general implementation of replace() will do).